tests/callbacks/test_tqdm_progress_bar.py

```diff
@@ -282,48 +271,48 @@ def test_tqdm_progress_bar_value_on_colab(tmpdir):
     assert trainer.progress_bar_callback.refresh_rate == 19
 
 
-class MockedUpdateProgressBars(TQDMProgressBar):
-    """Mocks the update method once bars get initializied."""
-
-    def _mock_bar_update(self, bar):
-        bar.update = Mock(wraps=bar.update)
-        return bar
+class MockTqdm(Tqdm):
+    def __init__(self, *args, **kwargs):
+        self.n_values = []
+        super().__init__(*args, **kwargs)
+        self.__n = 0
+        # again to reset additions from `super().__init__`
+        self.n_values = []
 
-    def init_train_tqdm(self):
-        bar = super().init_train_tqdm()
-        return self._mock_bar_update(bar)
+    @property
+    def n(self):
+        return self.__n
 
-    def init_validation_tqdm(self):
-        bar = super().init_validation_tqdm()
-        return self._mock_bar_update(bar)
-
-    def init_test_tqdm(self):
-        bar = super().init_test_tqdm()
-        return self._mock_bar_update(bar)
+    @n.setter
+    def n(self, value):
+        self.__n = value
+        # track the changes in the `n` value
+        if not len(self.n_values) or value != self.n_values[-1]:
+            self.n_values.append(value)
 
 
 @pytest.mark.parametrize(
-    "train_batches,val_batches,refresh_rate,train_deltas,val_deltas",
+    "train_batches,val_batches,refresh_rate,train_updates,val_updates",
     [
-        [2, 3, 1, [1, 1, 1, 1, 1], [1, 1, 1]],
-        [0, 0, 3, [], []],
-        [1, 0, 3, [1], []],
+        [2, 3, 1, [1, 2, 3, 4, 5], [1, 2, 3]],
+        [0, 0, 3, None, None],
+        [1, 0, 3, [1], None],
         [1, 1, 3, [2], [1]],
-        [5, 0, 3, [3, 2], []],
-        [5, 2, 3, [3, 3, 1], [2]],
-        [5, 2, 6, [6, 1], [2]],
+        [5, 0, 3, [3, 5], None],
+        [5, 2, 3, [3, 7], [2]],
+        [5, 2, 6, [7], [2]],
```

Comment:
Can you explain me the values there ? Not sure to understand where the 7 is coming from.

Reply:
We had 6 + 1 before because deltas were used. Deltas as in the difference between the previous and current call.

Now it's one update with the value 7 (6 + 1)

Reply:
Hey @carmocca.

I am not sure to fully understand your explanation.

Here is what I understand.

train_updates = [all the values divisible by train_batches] + [train_batches + val_batches]

[5, 2, 3] -> [3] + [5 + 2]
[5, 2, 6] -> [] (none is divisble) + [5 + 2]

Reply:
I don't get your point entirely, but this PR does not touch the refresh rate logic, it's still:

`(idx % self.refresh_rate == 0)` where the same `idx` values are passed.

This 7 here comes from the update `on_train_epoch_end`, where both training and validation have run (5 + 2 = 7)